src/windows/wslaservice/inc/wslaservice.idl

```diff
@@ -117,9 +201,41 @@ struct _WSLA_SESSION_SETTINGS {
 ]
 interface IWSLASession : IUnknown
 {
+    // Image management.
+    HRESULT PullImage([in] LPCWSTR Image, [in, unique] struct WSLA_REGISTRY_AUTHENTICATION_INFORMATION* RegistryInformation, [in, unique] IProgressCallback* ProgressCallback);
+    HRESULT ListImages([out, size_is(, *Count)] struct WSLA_IMAGE_INFORMATION** Images, [out] ULONG* Count);
+    HRESULT DeleteImage([in] LPCWSTR Image);
+
+    // Container management.
+    HRESULT CreateContainer([in] struct WSLA_CONTAINER_OPTIONS* Options);
```

Comment:
Have we considered returning a "container" as its own distinct COM object then just having interface methods that do this stuff. This would avoid the need to pass around the right container name and enforcing unique names.

Reply:
We do need to enforce unique names, because we want containers within a session to be able to refer to each other by names. 

I did consider having a dedicated COM object for containers, but when I designed it, it ended up adding an extra layer of object that the caller has to keep track off and required additional methods (like OpenContainer()) so that approach ended up being easier